src/Raven.Server/Documents/Handlers/AI/Agents/AbstractAiAgentProcessor.cs

```diff
@@ -103,6 +104,20 @@ public override async ValueTask ExecuteAsync()
                     throw new DocumentDoesNotExistException(conversationId);
 
                 conversationDocument = ConversationDocument.ToDocument(conversationId, conversation.Data);
+
+                if (changeVector != null)
+                {
+                    if (conversation.ChangeVector != changeVector)
+                        throw new ConcurrencyException($"The conversation '{conversationId}' was changed, please try again")
```

Comment:
Let's try to give the use a better message.

`The conversation '{conversationId}' was updated and doesn't match the expected change vector. Reload the conversation and try again.`